test/tests/functional/pbs_reservations.py

```diff
@@ -1487,45 +1475,58 @@ def test_job_array_within_recurring_reservation(self):
                            id=jids[1], extend='t')
         self.server.expect(JOB, {'job_state=Q': 3}, count=True,
                            id=jids[2], extend='t')
-        # At end of first instance of resrvation  ,verify running sub-job
-        # terminated
+        # At end of first instance of reservation ,verify running sub-job
+        # should be finished
         self.logger.info(
-            'Waiting 60 seconds for first  instance of reservation to end')
-        time.sleep(62)
-        job_list = subjid
-        job_list.pop()
-        job_list.pop()
-        for subjob in job_list:
-            self.server.expect(JOB, {'job_state': 'F', 'substate': '93'},
-                               extend='x', id=subjob)
+            'Waiting 30 sec job-array 1 and 2 to be finished')
+        self.server.expect(JOB, {'job_state=F': 3}, extend='xt',
+                           offset=30, id=jids[0])
+        self.server.expect(JOB, {'job_state=F': 3}, extend='xt',
+                           id=jids[1])
 
         # Wait for standing reservation second instance to confirmed
         exp_attr = {'reserve_state': (MATCH_RE, "RESV_CONFIRMED|2")}
-        self.server.expect(RESV, exp_attr, id=rid, offset=5)
-        # Check for queued jobs in second instance of resrvation
+        self.server.expect(RESV, exp_attr, id=rid)
+        # Check for queued jobs in second instance of reservation
         self.server.expect(JOB, {'job_state': 'Q',
                                  'comment': (MATCH_RE, 'Queue not started')},
                            id=jids[2])
-        self.server.expect(JOB, {'job_state=Q': 3}, extend='xt',
-                           id=jids[2])
         self.logger.info(
-            'Waiting 60 seconds for second  instance of reservation to start')
-        self.server.expect(RESV, {'reserve_index': 2}, offset=60, id=rid)
+            'Waiting 50 sec for second instance of reservation to start')
         exp_attr = {'reserve_state': (MATCH_RE, "RESV_RUNNING|5")}
-        self.server.expect(RESV, exp_attr, id=rid)
+        self.server.expect(RESV, exp_attr, id=rid, offset=50, interval=10)
+        self.server.expect(RESV, {'reserve_index': 2}, id=rid)
         # Check for queued jobs should be running
         self.server.expect(JOB, {'job_state=R': 2}, extend='xt',
                            id=jids[2])
 
         # Check for running jobs in second instance should finished
+        self.logger.info(
+            'Waiting 30 sec for second instance of reservation to finished')
         self.server.expect(JOB, {'job_state=F': 3}, extend='xt',
-                           offset=62, id=jids[2])
+                           offset=30, id=jids[2])
 
-        # Wait for reservations to be finished
+        # Wait for reservation to be finished
         msg = "Que;" + rid_q + ";deleted at request of pbs_server@"
         self.server.log_match(msg, starttime=end, interval=20)
```

Comment:
Here too, consider reducing the interval to a second because it will not take very long to end the reservation.